lib/iris/tests/test_constraints.py

```diff
@@ -295,50 +296,241 @@ def load_match(self, files, constraints):
         return cubes
 
 
+@tests.skip_data
+class TestCubeExtract__names(TestMixin, tests.IrisTest):
+    def setUp(self):
+        TestMixin.setUp(self)
+        self.cube = iris.load_cube(self.theta_path)
+        # Expected names...
+        self.standard_name = "air_potential_temperature"
+        self.long_name = "air potential temperature"
+        self.var_name = "apt"
+        self.stash = "m01s00i004"
+        # Configure missing names...
+        self.cube.long_name = self.long_name
+        self.cube.var_name = self.var_name
+
+    def test_standard_name(self):
+        constraint = iris.Constraint(self.standard_name)
+        result = self.cube.extract(constraint)
+        self.assertIsNotNone(result)
+        self.assertEqual(result.standard_name, self.standard_name)
```

Comment:
Not sure why we care about this last check ?
As we only have one cube, it doesn't do much

Reply:
@pp-mo I've change this somewhat, in that now the `setUp` contains more than one cube to extract against. In fact, this uncovered some issues that are now resolved, so I guess it's always good to extend tests... you just never know what you'll find :wink: 

So the above `assert` pattern is ensuring that "something" is returned, and also that what is returned is what we expected. It's a valid pattern that make sense now. Hope this helps.

Reply:
:+1: